x-pack/plugins/security_solution/server/feature_usage/service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LicensingPluginSetup, LicensingPluginStart } from '../../../licensing/server';
+
+const FEATURES = {
+  HOST_ISOLATION: 'Host isolation',
+  HOST_ISOLATION_EXCEPTION: 'Host isolation exception',
+  HOST_ISOLATION_EXCEPTION_BY_POLICY: 'Host isolation exception by policy',
+  TRUST_APP_BY_POLICY: 'Trusted app by policy',
+  EVENT_FILTERS_BY_POLICY: 'Event filters by policy',
+  RANSOMWARE_PROTECTION: 'Ransomeware protection',
+  MEMORY_THREAT_PROTECTION: 'Memory threat protection',
+  BEHAVIOR_PROTECTION: 'Behavior protection',
+};
+
+export class FeatureUsageService {
+  private licensingPluginStart: LicensingPluginStart | undefined;
+
+  public setup(licensingPluginSetup: LicensingPluginSetup) {
+    Object.values(FEATURES).map((featureValue) =>
+      licensingPluginSetup.featureUsage.register(featureValue, 'platinum')
+    );
+  }
+
+  public start(licensingPluginStart: LicensingPluginStart) {
+    this.licensingPluginStart = licensingPluginStart;
+  }
+
+  public notifyUsage() {
+    const notify = this.licensingPluginStart?.featureUsage?.notifyUsage || function () {};
```

Comment:
licensingPluginStart should never be undefined but since it can only be set in the service during plugin start, we can't initialize it in constructor so typescript. I considered using `!` here but never say never 😅 .

correct on `function () {}` for noop. I don't think it's worth throwing since it doesn't functionality break any features. good catch on `licensingPluginStart.featureUsage`, it's not optional so will remove extra check.